agent-inject/agent/config_test.go

```diff
@@ -766,3 +766,108 @@ func TestConfigAgentQuit(t *testing.T) {
 		})
 	}
 }
+
+func TestConfigTelemetry(t *testing.T) {
+	tests := []struct {
+		name              string
+		annotations       map[string]string
+		expectedTelemetry *Telemetry
+	}{
+		{
+			"annotations that exercise all of the annotations",
+			map[string]string{
+				"vault.hashicorp.com/agent-telemetry-usage_gauge_period":                     "10m",
+				"vault.hashicorp.com/agent-telemetry-maximum_gauge_cardinality":              "500",
+				"vault.hashicorp.com/agent-telemetry-disable_hostname":                       "false",
+				"vault.hashicorp.com/agent-telemetry-enable_hostname_label":                  "false",
+				"vault.hashicorp.com/agent-telemetry-lease_metrics_epsilon":                  "1h",
+				"vault.hashicorp.com/agent-telemetry-num_lease_metrics_buckets":              "168",
+				"vault.hashicorp.com/agent-telemetry-add_lease_metrics_namespace_labels":     "false",
+				"vault.hashicorp.com/agent-telemetry-filter_default":                         "true",
+				"vault.hashicorp.com/agent-telemetry-statsite_address":                       "https://foo.com",
+				"vault.hashicorp.com/agent-telemetry-statsd_address":                         "https://foo.com",
+				"vault.hashicorp.com/agent-telemetry-circonus_api_token":                     "foo",
+				"vault.hashicorp.com/agent-telemetry-circonus_api_app":                       "nomad",
+				"vault.hashicorp.com/agent-telemetry-circonus_api_url":                       "https://api.circonus.com/v2",
+				"vault.hashicorp.com/agent-telemetry-circonus_submission_interval":           "10s",
+				"vault.hashicorp.com/agent-telemetry-circonus_submission_url":                "https://api.circonus.com/v2",
+				"vault.hashicorp.com/agent-telemetry-circonus_check_id":                      "foo",
+				"vault.hashicorp.com/agent-telemetry-circonus_check_force_metric_activation": "false",
+				"vault.hashicorp.com/agent-telemetry-circonus_check_instance_id":             "foo:bar",
+				"vault.hashicorp.com/agent-telemetry-circonus_check_search_tag":              "foo:bar",
+				"vault.hashicorp.com/agent-telemetry-circonus_check_display_name":            "foo",
+				"vault.hashicorp.com/agent-telemetry-circonus_check_tags":                    "foo,bar",
+				"vault.hashicorp.com/agent-telemetry-circonus_broker_id":                     "foo",
+				"vault.hashicorp.com/agent-telemetry-circonus_broker_select_tag":             "foo:bar",
+				"vault.hashicorp.com/agent-telemetry-dogstatsd_addr":                         "https://foo.com",
+				"vault.hashicorp.com/agent-telemetry-dogstatsd_tags":                         "[\"foo:bar\", \"foo:baz\"]",
```

Comment:
nit: we typically prefer using back quotes instead of escaping the `"`, whenever possible. 

```suggestion
				"vault.hashicorp.com/agent-telemetry-dogstatsd_tags":                         `["foo:bar", "foo:baz"]`,
```